src/shared/json.c

```diff
@@ -1453,6 +1456,29 @@ bool json_variant_is_sensitive(JsonVariant *v) {
         return v->sensitive;
 }
 
+bool json_variant_is_sensitive_recursive(JsonVariant *v) {
+        if (!v)
+                return false;
+        if (json_variant_is_sensitive(v))
+                return true;
+        if (!json_variant_is_regular(v))
+                return false;
+        if (v->recursive_sensitive) /* Already checked this before */
+                return true;
+        if (!IN_SET(v->type, JSON_VARIANT_ARRAY, JSON_VARIANT_OBJECT))
+                return false;
+        if (v->is_reference)
+                return (v->recursive_sensitive = json_variant_is_sensitive_recursive(v->reference));
```

Comment:
This caches the result even if it's false, is that intended?

Reply:
well, that has no effect. we check whether the field is true a few lines earlier, and if so immediately return. so we know here that the field is false. Thus in the worst case we might override false with false, which is a a NOP.

these makes the sources shorter, that's why i did it this way.